### PR TITLE
Use repline for DAML REPL

### PR DESCRIPTION
**Pull request by aherrmann-da**
This provides readline bindings and history

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/ReplServiceMain.scala [hunk 1]
```diff
@@ -216,7 +216,9 @@ class ReplService(
       ApplicationId("daml repl"),
       TimeProvider.UTC)
     runner.runWithClients(clients).onComplete {
-      case Failure(e) => respObs.onError(e)
+      case Failure(e) =>
+        System.err.println(s"$e")
+        respObs.onError(e)
```
Comment (aherrmann-da) [inline compiler/repl-service/server/src/main/scala/com/digitalasset/daml/lf/ReplServiceMain.scala:221]: The REPL server was not printing errors such as when trying to re-allocate the same party. I'm working on a test-case for a separate PR.

File: compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs [hunk 2]
```diff
@@ -168,7 +175,21 @@ runRepl opts mainDar replClient ideState = do
                 hPutStrLn stderr ("Package could not be loaded: " <> show err)
                 exitFailure
             Right _ -> pure ()
-    go moduleNames 0 []
+    let initReplState = ReplState
+          { moduleNames = moduleNames
+          , bindings = []
+          , lineNumber = 0
+          }
+    -- TODO[AH] Use Repl.evalReplOpts once we're using repline >= 0.2.2
```
Comment (cocreature) [inline compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs:183]: Can we upgrade already?

Comment (aherrmann-da) [reply compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs:183]: Sadly not, there's a version constraint on `haskeline` and that's a core package. So, this is blocked on a GHC upgrade.

Comment (cocreature) [reply compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs:183]: Ah right :disappointed:

**Pull request merged**